conan-center-index is currently not able to correctly the components of HDF5.

--- liblib/CMakeLists.txt
+++ liblib/CMakeLists.txt
@@ -81,7 +81,7 @@
     # builds:
     # Make sure that HDF5_C_LIBRARY appears *after*
     # HDF5_HL_LIBRARY.
-    SET(TLL_LIBS ${HDF5_HL_LIBRARIES} ${HDF5_C_LIBRARIES} ${TLL_LIBS} ${SZIP_LIBRARY})
+    SET(TLL_LIBS ${HDF5_HL_LIBRARIES} ${HDF5_LIBRARIES} ${TLL_LIBS} ${SZIP_LIBRARY})
   ELSE() # Windows CMake defines HDF5_LIBRARIES.
     SET(TLL_LIBS ${HDF5_LIBRARIES} ${TLL_LIBS} ${SZIP_LIBRARY})
   ENDIF()
